ouroboros-consensus/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -124,6 +128,127 @@ prop_Mempool_InvalidTxsNeverAdded setup =
         , txInMempool `notElem` txsInMempoolBefore
         ]
 
+-- | When the mempool is at capacity, test that 'addTxs' blocks when
+-- attempting to add one more transaction and that it unblocks when there is
+-- adequate space. Adequate space is made by adding all of the existing
+-- transactions to the ledger and removing them from the mempool.
+prop_Mempool_Capacity :: Property
+prop_Mempool_Capacity =
+  let nbInitialTxs = 0
+  in  forAll (choose (2, 100)) $ \nbNewTxs ->
```

Comment:
I'm not a fan of `forAll`s inside the body:
* No shrinking
* You can't reproduce a failing test case by just passing the arguments (that QC prints) to it.

So I suggest creating a data type containing everything you need and writing an `Arbitrary` instance for that.

Reply:
Good point. I'll do that.